File: 백준/Silver/2870. 수학숙제/수학숙제.scala
import scala.io.StdIn.readLine

object Main {
    def main(args: Array[String]): Unit = {
        val n = readLine.toInt
        (1 to n).foldLeft(List[BigInt]())((acc, _) => {
            val str = readLine.split("[a-z]")
                .filter(_.nonEmpty)
                .map(num => BigInt(num))
                .toList
            acc.concat(str)
        }).sorted.foreach(println)
    }
}

File: 백준/Silver/2870. 수학숙제/수학숙제.java
import java.util.*;
import java.io.*;
import java.math.BigInteger;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        List<BigInteger> l = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            Arrays.stream(br.readLine().split("[a-z]+"))
                .filter(s -> !s.isEmpty())
                .forEach(s -> l.add(new BigInteger(s)));
        }
        l.stream().sorted().forEach(System.out::println);
    }
}